pkg/pillar/agentlog/loglevel.go

```diff
@@ -69,98 +75,101 @@ func getRemoteLogLevelImpl(sub *pubsub.Subscription, agentName string,
 		log.Infof("GetRemoteLogLevel failed %s\n", err)
 		return "", false
 	}
-	gc := cast.CastGlobalConfig(m)
+	gc := cast.CastConfigItemValueMap(m)
 	// Do we have an entry for this agent?
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.RemoteLogLevel != "" {
-		return as.RemoteLogLevel, true
-	}
-	// Do we have a default value?
-	if allowDefault && gc.DefaultRemoteLogLevel != "" {
-		return gc.DefaultRemoteLogLevel, true
+	if _, ok1 := gc.AgentSettings[agentName]; ok1 {
+		as, ok := gc.AgentSettings[agentName]["remote.loglevel"]
+		if ok && as.StringValue != "" {
+			return as.Key, true
+		} else if ok && as.StringValue != "" && allowDefault {
+			// Return default remote.loglevel (global)
+			return "global", true
+		} else {
+			log.Infof("Failed to find loglevel settings for %s", agentName)
+		}
+	} else {
+		log.Infof("Failed to find per agent settings for %s", agentName)
 	}
 	return "", false
 }
 
-func LogLevel(gc *types.GlobalConfig, agentName string) string {
-
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.LogLevel != "" {
-		return as.LogLevel
+func LogLevel(gc *types.ConfigItemValueMap, agentName string) string {
```

Comment:
There is getLogLevelImpl , GetLogLevel functions - Is this supposed to override them? 

The logLevel logic I mentioned is better put in here.. isnt it??

Reply:
This function existed before. I didn't make it. Should I remove it? The project does not use.

Reply:
If this function is unused, remove it..